routing/routes.go

```diff
@@ -2,33 +2,69 @@ package router
 
-	router.HEAD("/ops/ping", opsHandler.PingHandler)
-	router.GET("/ops/status", opsHandler.StatusHandler)
+	router.HandleFunc("/ops/ping", opsHandler.PingHandler).Methods("HEAD")
+	router.HandleFunc("/ops/status", opsHandler.StatusHandler).Methods("GET")
 
-	router.GET("/dataset", metaHandler.GetDatasets)
-	router.GET("/timeseries", metaHandler.GetTimeseries)
+	router.HandleFunc("/dataset", metaHandler.GetDatasets).Methods("GET")
+	router.HandleFunc("/dataset/{datasetId}", metaHandler.GetSpecificDataset).Methods("GET")
+	router.HandleFunc("/dataset/{datasetId}/timeseries", metaHandler.GetSpecificDatasetTimeSeries).Methods("GET")
+	router.HandleFunc("/dataset/{datasetId}/timeseries/{timeseriesId}", metaHandler.GetSpecificTimeSeriesSpecificDataset).Methods("GET")
+	router.HandleFunc("/dataset/{datasetId}/timeseries/{timeseriesId}/data", dataHandler.GetData).Methods("GET")
 
-	router.GET("/dataset/:datasetId", metaHandler.GetSpecificDataset)
-	router.GET("/timeseries/:timeseriesId", metaHandler.GetSpecificTimeseries)
+	router.HandleFunc("/timeseries", metaHandler.GetTimeseries).Methods("GET")
+	router.HandleFunc("/timeseries/{timeseriesId}", metaHandler.GetSpecificTimeseries).Methods("GET")
+	router.HandleFunc("/timeseries/{timeseriesId}/dataset", metaHandler.GetSpecificTimeseriesDatasets).Methods("GET")
+	router.HandleFunc("/timeseries/{timeseriesId}/dataset/{datasetId}", metaHandler.GetSpecificTimeSeriesSpecificDataset).Methods("GET")
+	router.HandleFunc("/timeseries/{timeseriesId}/dataset/{datasetId}/data", dataHandler.GetData).Methods("GET")
 
-	router.GET("/dataset/:datasetId/timeseries", metaHandler.GetSpecificDatasetTimeSeries)
-	router.GET("/timeseries/:timeseriesId/dataset", metaHandler.GetSpecificTimeseriesDatasets)
+	router.HandleFunc("/search", metaHandler.DoSearch).Methods("GET")
 
-	router.GET("/dataset/:datasetId/timeseries/:timeseriesId", metaHandler.GetSpecificTimeSeriesSpecificDataset)
-	router.GET("/timeseries/:timeseriesId/dataset/:datasetId", metaHandler.GetSpecificTimeSeriesSpecificDataset)
+	return router
+}
 
-	router.GET("/search", metaHandler.DoSearch)
+func DeprecationMiddleware(deprecation config.Deprecation) func(http.Handler) http.Handler {
+	return func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 
-	router.GET("/dataset/:datasetId/timeseries/:timeseriesId/data", dataHandler.GetData)
-	router.GET("/timeseries/:timeseriesId/dataset/:datasetId/data", dataHandler.GetData)
+			if deprecation.Sunset != "" {
+				sunsetTime, err := time.Parse(time.RFC1123, deprecation.Sunset)
+				if err != nil {
+					log.Warn(req.Context(), "failing to parse configuration of Sunset header needed for service deprecation, continue allowing users access to endpoints", log.FormatErrors([]error{err}))
+				}
 
-	return router
+				if sunsetTime.Before(time.Now().UTC()) {
+					w.WriteHeader(http.StatusNotFound)
+					return
+				}
+			}
+
+			if deprecation.IsDeprecated {
```

Comment:
confused that you can use `IsDeprecated == false` yet when `Sunset != ""` you still get 404 (eventually)

need confirmation, please (i.e. is Sunset confusingly inside Deprecation struct when it shouldn't be?)

Reply:
So I've not locked down all combinations as I assume the configuration setup will be correctly used. Sunset value should always be set after or at the same time as deprecation value.

I could lock it down and ignore sunset header if deprecation header is not set.

Happy to have a chat

Reply:
`hasDeprecationHeader`?